### PR TITLE
Settings

### PR DESCRIPTION
**Pull request by Harinze**

### CODE REVIEW

**Review by Don-christo — CHANGES_REQUESTED** [review]
Good job. LGTM! Please, go through the comments, and tell me what you think, then make the little changes if necessary. After that, re-request my review so that I can approve for merging. Thank you.

File: apps/server/src/controllers/riderControllers/editProfileSettings.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import { Request, Response } from "express";
+import Ryder from "../../models/ryder";
+import logger from "../../utilities/logger";
+import { registerSchema } from "../../utilities/validators";
+import { HTTP_STATUS_CODE } from "../../constants/httpStatusCode";
+
+export const editProfile = async (req: Request, res: Response) => {
+  const userId = req.params.userId;
```
[the remark this answers is not in the document]
Comment (Harinze) [reply apps/server/src/controllers/riderControllers/editProfileSettings.ts:8]: userId is preferred to id.

File: apps/server/src/controllers/riderControllers/editProfileSettings.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import { Request, Response } from "express";
+import Ryder from "../../models/ryder";
+import logger from "../../utilities/logger";
+import { registerSchema } from "../../utilities/validators";
+import { HTTP_STATUS_CODE } from "../../constants/httpStatusCode";
+
+export const editProfile = async (req: Request, res: Response) => {
```
[the remark this answers is not in the document]
Comment (Harinze) [reply apps/server/src/controllers/riderControllers/editProfileSettings.ts:7]: we are working from ryder dir, pls. u know we have customer dir and ryder dir

File: apps/server/src/routes/v1/riders.ts [hunk 3]
```diff
@@ -1,9 +1,12 @@
 import { Router } from "express";
+import {editProfile } from "../../controllers/riderControllers/editProfileSettings"
+import { auth } from "../../middleware/authorization";
 import { registerRyder, getRiders } from "../../controllers/riderControllers";
 
 const router = Router();
 
 router.post("/registerRider", registerRyder);
+router.put("/editprofile/:userId",  auth, editProfile);
```
[the remark this answers is not in the document]
Comment (Harinze) [reply apps/server/src/routes/v1/riders.ts:10]: userId is preferred

File: apps/Client/src/App.tsx [hunk 4]
```diff
@@ -11,6 +12,7 @@ function App() {
         <Route path="/" element={<HomePage />} />
         <Route path="/request-rider" element={<RequestRiderPage />} />
         <Route path="/signup/rider" element={<RiderSignUpPage />} />
+        <Route path="/userprofilesettings" element={<UserprofileSettings />} />
```
[the remark this answers is not in the document]
Comment (Harinze) [reply apps/Client/src/App.tsx:15]: have you seen a url that is camelcased. urls are most of times small cased.   read solid principles ..lolz

File: apps/Client/src/routes/Routes.tsx [hunk 5]
```diff
@@ -4,12 +4,14 @@ import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
 import RequestRiderPage from '../pages/RequestRiderPage';
 
 
+
 const AppRoutes: React.FC = () => {
   return (
     <Router>
       <Routes>
         <Route path="/request-rider" element={<RequestRiderPage />} />
```
[the remark this answers is not in the document]
Comment (Harinze) [reply apps/Client/src/routes/Routes.tsx:12]: url is small cased

File: apps/Client/src/App.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (Harinze) [reply apps/Client/src/App.tsx:15]: UserprofileSettings: this is camelcased

File: apps/server/src/controllers/riderControllers/editProfileSettings.ts [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+    res.status(HTTP_STATUS_CODE.SUCCESS).json({ message: 'Profile updated successfully', user:{
+      firstName: user.firstName,
+      lastName: user.lastName,
+      email: user.email,
+      phone: user.phone,
+      userId: user.id
+    } });
+  } catch (error) {
+    console.error(error);
```
[the remark this answers is not in the document]
Comment (Harinze) [reply apps/server/src/controllers/riderControllers/editProfileSettings.ts:41]: ok

File: apps/server/src/routes/v1/riders.ts [hunk 7]
```diff
@@ -1,9 +1,12 @@
 import { Router } from "express";
+import {editProfile } from "../../controllers/riderControllers/editProfileSettings"
```
[the remark this answers is not in the document]
Comment (Don-christo) [reply apps/server/src/routes/v1/riders.ts]: Yes, that is what you did. I guess you didn't understand what I suggested.

**Review by Don-christo — APPROVED** [review]
Good to go! Well done.

**Pull request merged**